onnxscript/_framework_apis/torch_2_5.py

```diff
@@ -68,32 +67,16 @@ def save_model_with_external_data(model: ir.Model, model_path: str | os.PathLike
     """Save the model with external data. The model is unchanged after saving."""
 
     # TODO(#1835): Decide if we want to externalize large attributes as well
-    initializer_values = tuple(model.graph.initializers.values())
-    tensors = [v.const_value for v in initializer_values]
-    for tensor in tensors:
-        if tensor is None:
+    for value in model.graph.initializers.values():
+        if value.const_value is None:
```

Comment:
Maybe it is worth catching this when the value is added to the model not when it is saved. That way, the user has more info about when he did something wrong.

Reply:
Sounds good. For that we will change pytorch exporter to add the message

Reply:
I guess this is an internal invariant? I am guessing the only reason it can fail is if some user explicitly sets this value to be None via Value's setter method?